src/main/kotlin/cacophony/grammar/syntaxtree/Expression.kt

```diff
@@ -0,0 +1,253 @@
+package cacophony.grammar.syntaxtree
+
+import cacophony.utils.Location
+
+sealed class Expression(
+    val range: Pair<Location, Location>,
+) { // everything in cacophony is an expression
+    // artificial instance, can be useful when calculating values of nested expressions
+    class Empty(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class VariableUse(
+        range: Pair<Location, Location>,
+        val identifier: String,
+    ) : Expression(range)
+
+    sealed class Type(
+        range: Pair<Location, Location>,
+    ) {
+        class Basic(
+            range: Pair<Location, Location>,
+            val identifier: String,
+        ) : Type(range)
+
+        class Functional(
+            range: Pair<Location, Location>,
```

Comment:
Move the Type class outside of the Expression class (so that it is at the top level). Maybe make range a val field?
```Kotlin
sealed class Type(
    val range: Pair<Location, Location>,
) {
    class Basic(
        range: Pair<Location, Location>,
        val identifier: String,
    ) : Type(range)

    class Functional(
        range: Pair<Location, Location>,
        vararg val argumentsType: Type,
        val returnType: Type,
    ) : Type(range)
}
```